contract/src/DEAL.sol

```diff
@@ -153,33 +120,48 @@ contract DEAL is ERC20 {
     function burn(address token, uint256 burnAmount) public {
         require(initialised, "!initialised");
         require(reserves[token].token != address(0), "unsupported token");
-        
-        uint256 e = e_calc(burnAmount);
-        uint256 amountToBurnE18 = totalReserveBalance - (totalReserveBalance * 1e18) / e;
-        
+
+        uint256 amountToBurnE18 = computeBurnAmount(burnAmount);
+
         // Convert to token decimals and check if we have enough
         uint256 tokenAmount = _fromE18(amountToBurnE18, reserves[token].decimals);
         require(reserves[token].balance >= tokenAmount, "insufficient liquidity");
-        
+
         _burn(msg.sender, burnAmount);
         totalReserveBalance -= amountToBurnE18;
         reserves[token].balance -= tokenAmount;
-        
+
         SafeTransferLib.safeTransfer(ERC20(token), msg.sender, tokenAmount);
-        emit DealBurned(msg.sender, burnAmount, tokenAmount, token, e);
+        emit DealBurned(msg.sender, burnAmount, tokenAmount, token);
+    }
+
+    function computeBurnAmount(uint256 burnAmount) view internal returns (uint256 result){
+        PRBMath.UD60x18 memory kUD = PRBMathUD60x18.fromUint(K); // k
+        PRBMath.UD60x18 memory burnAmountUD = PRBMathUD60x18.fromUint(burnAmount); // T
+        PRBMath.UD60x18 memory trUD = PRBMathUD60x18.fromUint(totalReserveBalance);
+        PRBMath.UD60x18 memory scalingFactorUD = PRBMathUD60x18.fromUint(SCALING_FACTOR);
+        PRBMath.UD60x18 memory rScalingFactorUD = PRBMathUD60x18.mul(scalingFactorUD, trUD); // R * (SCALING_FACTOR)
+        PRBMath.UD60x18 memory xUD = PRBMathUD60x18.div(rScalingFactorUD, PRBMathUD60x18.exp(PRBMathUD60x18.div(burnAmountUD, kUD))); // R * (SCALING_FACTOR) / e^{T/k}
+        PRBMath.UD60x18 memory resultUD = PRBMathUD60x18.sub(trUD, xUD); // (R - R/e^{T/k}) 
+        result = PRBMathUD60x18.toUint(resultUD); 
     }
 
-    // Original helper functions remain unchanged
-    function e_calc(uint256 x) internal pure returns (uint256 result) {
```

Comment:
We're going to need to talk this one through.

1. I don't like that a serous change to the maths is included in a PR with lots of other formatting stuff. They should be separate so we can think clearly about the specific, functional change and nothing else.
2. It's unclear why you think we should case everything into PRB first and add many more lines of code here, with additional functions. Let's talk about this.